backend/app/crud/user.py

```
@@ -0,0 +1,64 @@
+        return False
+
+    db.delete(db_user)
+    db.commit()
+    return True
+
+
+def authenticate_user(db: Session, email: str, password: str) -> Optional[User]:
+    user = get_user_by_email(db, email)
+    if not user:
+        return None
+    hashed_password = getattr(user, "hashed_password")
```

Ruff (0.8.2)

Do not call `getattr` with a constant attribute value. It is not any safer than normal property access.

Replace `getattr` with attribute access

(B009)